### PR TITLE
Consolidate MixinVaultCore and ZrxVault + unit tests

### PR DESCRIPTION
**Pull request by mzhu25**
## Description
Since we're down to one vault, might as well move the stuff in `MixinVaultCore` into `ZrxVault`. This PR also includes unit tests for the newly consolidated `ZrxVault`, as well as cosmetic changes to other test files. 

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/staking/test/unit_tests/zrx_vault_test.ts [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+        // configure erc20 proxy to accept calls from zrx vault
+        await erc20ProxyContract.addAuthorizedAddress.awaitTransactionSuccessAsync(zrxVault.address);
+    });
+
+    enum ZrxTransfer {
+        Deposit,
+        Withdrawal,
+    }
+
+    async function verifyTransferPostconditionsAsync(
```
Comment (jalextowle) [inline contracts/staking/test/unit_tests/zrx_vault_test.ts]: Naming level 💯

File: contracts/staking/test/unit_tests/zrx_vault_test.ts [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+        transferType: ZrxTransfer,
+        staker: string,
+        amount: BigNumber,
+        initialVaultBalance: BigNumber,
+        initialTokenBalance: BigNumber,
+        receipt: TransactionReceiptWithDecodedLogs,
+    ): Promise<void> {
+        const eventArgs =
+            transferType === ZrxTransfer.Deposit
```
Comment (jalextowle) [inline contracts/staking/test/unit_tests/zrx_vault_test.ts]: This is super clean 😤

File: contracts/staking/test/unit_tests/zrx_vault_test.ts [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+                expect(tx).to.revertWith(expectedError);
+            });
+            it('Owner can set the staking proxy', async () => {
+                const newAddress = nonOwnerAddresses[0];
+                const receipt = await zrxVault.setStakingProxy.awaitTransactionSuccessAsync(newAddress, {
+                    from: owner,
+                });
+                await verifyStakingProxySetAsync(receipt, newAddress);
+            });
+            it('Authorized address can set the staking proxy', async () => {
```
Comment (jalextowle) [inline contracts/staking/test/unit_tests/zrx_vault_test.ts]: If we make `setZrxProxy` authorized, then we should add tests like this for that function.

File: contracts/staking/test/unit_tests/zrx_vault_test.ts [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+                    await verifyTransferPostconditionsAsync(
+                        ZrxTransfer.Deposit,
+                        staker,
+                        new BigNumber(1),
+                        initialVaultBalance,
+                        initialTokenBalance,
+                        receipt,
+                    );
+                });
+                it('Staking proxy can deposit entire ZRX balance on behalf of staker', async () => {
```
Comment (jalextowle) [inline contracts/staking/test/unit_tests/zrx_vault_test.ts]: 🙏

File: contracts/staking/test/unit_tests/zrx_vault_test.ts [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+                        staker,
+                        initialVaultBalance,
+                        initialVaultBalance,
+                        initialTokenBalance,
+                        receipt,
+                    );
+                });
+                it('Someone can withdraw ZRX on behalf of a staker', async () => {
+                    const receipt = await zrxVault.withdrawAllFrom.awaitTransactionSuccessAsync(staker, {
+                        from: owner,
```
Comment (jalextowle) [inline contracts/staking/test/unit_tests/zrx_vault_test.ts]: Nit: I think we should also check with a non-owner.

File: contracts/staking/contracts/src/ZrxVault.sol [hunk 6]
```diff
@@ -114,9 +153,10 @@ contract ZrxVault is
     /// @param amount of Zrx Tokens to withdraw.
     function withdrawFrom(address staker, uint256 amount)
         external
-        onlyStakingProxy
-        onlyNotInCatastrophicFailure
     {
+        _assertSenderIsStakingProxy();
```
[the remark this answers is not in the document]
Comment (abandeali1) [reply contracts/staking/contracts/src/ZrxVault.sol]: +1

File: contracts/staking/contracts/src/ZrxVault.sol [hunk 7]
```diff
@@ -69,15 +84,38 @@ contract ZrxVault is
...
     /// @dev Sets the Zrx proxy.
-    /// Note that only an authorized address can call this function.
+    /// Note that only the contract owner can call this.
     /// Note that this can only be called when *not* in Catastrophic Failure mode.
     /// @param _zrxProxyAddress Address of the 0x Zrx Proxy.
     function setZrxProxy(address _zrxProxyAddress)
         external
-        onlyAuthorized
-        onlyNotInCatastrophicFailure
+        onlyOwner
```
[the remark this answers is not in the document]
Comment (mzhu25) [reply contracts/staking/contracts/src/ZrxVault.sol]: good catch, must've slipped through in a rebase at some point

**Pull request merged**